### ISSUE TITLE
v1.18.2 broken on OpenBSD/Illumos/Solaris, linked against nonexistent DSO - "Cannot find /lib64/ld-linux-x86-64.so.2"

### ISSUE THREAD

**Issue by whorfin** [syncthing/syncthing#7934]
Starting with an autoupdate September 7th at 05am PST, syncthing crashes with:
```
genunix: [ID 700856 kern.notice] syncthing: Cannot find /lib64/ld-linux-x86-64.so.2
```
I do not know which version of syncthing this was, as the log does not say and it can't be run; the string "Fermium Flea" is found in it

If I run either the version which autodownloaded or the version from the 1.18.2-rc.3
https://github.com/syncthing/syncthing/releases/download/v1.18.2-rc.3/syncthing-illumos-amd64-v1.18.2-rc.3.tar.gz

I get the same result:
```
$ ./syncthing --no-browser
syncthing: Cannot find /lib64/ld-linux-x86-64.so.2
```
Indeed just `ldd` fails:
```
$ ldd syncthing
syncthing: Cannot find /lib64/ld-linux-x86-64.so.2
```

This is on:
```
$ uname -a
SunOS serverbox 5.11 omnios-r151038-b3a82a213e i86pc i386 i86pc illumos
```

I note that the "releases" for 1.18.2 are missing an illumos build, rendering the syncthing official "Download" page currently broken for this OS.

`v1.18.1` seems to be OK, and has not been built against that bogus nonexistant library:
```
$ ldd syncthing-illumos-amd64-v1.18.1/syncthing
        libsendfile.so =>        /lib/64/libsendfile.so
        libsocket.so =>  /lib/64/libsocket.so
        libc.so =>       /lib/64/libc.so
        libnsl.so.1 =>   /lib/64/libnsl.so.1
        libmp.so.2 =>    /lib/64/libmp.so.2
        libmd.so.1 =>    /lib/64/libmd.so.1
        libm.so.2 =>     /lib/64/libm.so.2
```

`elfdump` shows the `.interp` section of the working version to point to `/lib/amd64/ld.so.1`, and the busted versions point to `/lib64/ld-linux-x86-64.so.2`

So running this evil command gave me a [temporarily] working syncthing:
```
$ patchelf --set-interpreter /lib/amd64/ld.so.1 syncthing
```

QED - there is some kind of build-system problem with illumos targets at the moment

**Comment by whorfin** [comment]
...after the `patchelf` shenanigans, the autoupdate-which-borked-everything version reports itself as:
```
Version | v1.18.2, Solaris (64-bit Intel/AMD)
```

**Comment by AudriusButkevicius** [comment]
See: https://forum.syncthing.net/t/latest-release-v1-18-2-doesnt-work-on-omnios-solaris/17318

**Comment by whorfin** [comment]
A working 1.18.2 can be made with the `patchelf` magic above.  I'll upload mine for anyone who wants it, disclaiming all fitness to purpose, it might murder your pets etc.  Don't download things from strangers, and so forth.
You could temporarily do the patchelf in the build, although clearly as I said there is something amiss in the build.
[syncthing-illumos-v1.18.2-fixed.tar.gz](https://github.com/syncthing/syncthing/files/7139381/syncthing-illumos-v1.18.2-fixed.tar.gz)

**Comment by calmh** [comment]
Yeah, as noted in the forum thread it's a Go bug. It'll likely be fixed in a point release before 1.18.3, otherwise we'll compile with Go 1.16 for the affected platforms. In the meantime I pulled the bad 1.18.2 binaries yesterday.
